charming/src/element/step.rs

```diff
@@ -0,0 +1,9 @@
+use serde::Serialize;
+
+#[derive(Serialize, Debug, PartialEq, PartialOrd, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum Step {
+    Start,
+    Middle,
+    End,
```

Comment:
I was just about to create a PR with the support for this same feature =)
We could've made it fully compatible with Echarts by adding bool variants:
```rust
pub enum Step {
    True,
    False,
    Start,
    Middle,
    End,
}

impl From<bool> for Step {
    fn from(value: bool) -> Self {
        if value {
            Step::True
        } else {
            Step::False
        }
    }
}
```

Reply:
Ah, although maybe I'm missing the serialization part, true and false would have to become boolean literals rather than strings, anyway, just a suggestion

Reply:
This would be correct. It should be fairly easy and examples of it are in the repo. 

I will take a look at this PR and merge it at the start of next week.

Reply:
@cooljingle Do you want to implement the suggestion from grinya, the implementation `From<bool>` and the correct serialization of `Step::True` and `Step::False`?

Reply:
@LukaOber just to confirm, you'd like the enum to retain the `True` and `False` variants, and have a custom serde `Serialize` implementation for it?

I'm not entirely sure of the philosophy of charming, but I'm thinking it might be a little confusing having all these variants in the Rust code. `impl From<bool>` seems reasonable, but having a play with echarts shows that `bool` is the same as `Start`, making the other variants `True` and `False` seem a little redundant.

Reply:
> @LukaOber just to confirm, you'd like the enum to retain the `True` and `False` variants, and have a custom serde `Serialize` implementation for it?

Yes, the fields should contain the five available options from echarts [series-line.step](https://echarts.apache.org/en/option.html#series-line.step). Echarts has two subtypes for this. 
Either the strings: `'start'`, `'middle'`, `'end'` 
Or the boolean options: `true`, `false`

We need a custom Serializer because the boolean options are not strings and need to be serialized as `true` and not as `'true'`

> I'm not entirely sure of the philosophy of charming, but I'm thinking it might be a little confusing having all these variants in the Rust code. `impl From<bool>` seems reasonable, but having a play with echarts shows that `bool` is the same as `Start`, making the other variants `True` and `False` seem a little redundant.

I am not sure if there is a difference between `true` and `'start'` but we should just be able to serialize all options correctly. 

If you have any more questions, I will be happy to help.

Reply:
@LukaOber thanks for finishing this off, glad to see the feature merged 👍